roadmap/views.py

```diff
@@ -1,15 +1,73 @@
-from rest_framework.generics import ListAPIView
+"""Roadmad related views"""
+from django.http import Http404
+from rest_framework import generics, status
 from rest_framework.response import Response
+from rest_framework.exceptions import NotAcceptable, NotFound
 
-from roadmap.models import EntryLevel, LocalDomain, MetaDomain
-from roadmap.serializers import (
-    EntryLevelSerializer,
-    LocalDomainSerializer,
-    MetaDomainSerializer,
-)
+from roadmap.models import Roadmap, EntryLevel, LocalDomain, MetaDomain
+from roadmap.serializers import (RoadmapDetailSerializer,
+                                 RoadmapSearchSerializer,
+                                 EntryLevelSerializer,
+                                 LocalDomainSerializer,
+                                 MetaDomainSerializer,)
+from roadmap.pagination import RoadmapSearchPagination
 
 
-class DomainListAPIView(ListAPIView):
+
+
+class RoadmapSearchAPIView(generics.ListAPIView):
+    serializer_class = RoadmapSearchSerializer
+    pagination_class = RoadmapSearchPagination
+
+    def get_queryset(self):
+        search_query = self.request.GET.get('search')
+
+        if not search_query:
+            raise NotAcceptable({
+                "message": "Parameter <search> required.",
+                "code": status.HTTP_406_NOT_ACCEPTABLE
```

Comment:
why do we return 406 and not 400 ? look what errors could be return in open api schema